From 9aca2bf97ac595c3006e6b6347493929263f3a32 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Alex=20R=C3=B8nne=20Petersen?= <alex@alexrp.com>
Date: Wed, 16 Jul 2025 20:20:09 +0200
Subject: [PATCH 070/133] compiler-rt: use native f16 type for loongarch
X-Developer-Signature: v=1; a=openpgp-sha256; l=606; i=xtex@astrafall.org;
 h=from:subject; bh=fUCa+jCxYqJjKzArH43SJ7N8pojALJZj4neZd9hBuf8=;
 b=owGbwMvMwCW2U4Ij7wZL9ETG02pJDJn78lgkVnuvcp+jvm7S+rDHWza++F6yPmDXpwOLnxvu0
 jLazKbM1TGRhUGMi8FSTJGlyLDBm1UnnV90WbkszBxWJpAh0iINDAwMDCwMfLmJeaVGOkZ6ptqG
 eoaGOkY6RgxcnAIw1bJ1DP8UfHa7fnl67GjRbM3Od29+L76zIFB12/R1a/8/unDp61k+a4b/2XY
 xdTeF8o7U9ttsnprzTuBWwbwDbqdrH9h4HmfMC7JgBwA=
X-Developer-Key: i=xtex@astrafall.org; a=openpgp;
 fpr=7231804B052C670F15A6771DB918086ED8045B91

This ABI changed in LLVM 21.

(cherry picked from commit c4592da88230939c068350e469bdc34076260860)
---
 lib/compiler_rt/common.zig | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/lib/compiler_rt/common.zig b/lib/compiler_rt/common.zig
index 1160b1c718e5..bae105cf4478 100644
--- a/lib/compiler_rt/common.zig
+++ b/lib/compiler_rt/common.zig
@@ -121,6 +121,8 @@ pub fn F16T(comptime OtherType: type) type {
         .thumbeb,
         .aarch64,
         .aarch64_be,
+        .loongarch32,
+        .loongarch64,
         .nvptx,
         .nvptx64,
         .riscv32,
-- 
2.52.0

